src/kernel/src/obj/range.rs

```diff
@@ -3,12 +3,93 @@ use core::fmt::Display;
 
 use nonoverlapping_interval_tree::{IntervalValue, NonOverlappingIntervalTree};
 
+//
+// KANI TODO
+//
+
+use crate::mutex::Mutex;
+
+
+//Includes Inline assembly in the dependencies. Not supported by kani
+//TODO: Currently runs out of memory :(
+#[cfg(kani)]
+mod page_range_tree_verification{
+    use core::arch::x86_64::CpuidResult;
+
+    use crate::memory::PhysAddr;
+    use crate::memory::VirtAddr;
+    use crate::obj::PageNumber;
+
+    use super::PageRangeTree;
+    use super::Page;
+    use twizzler_abi::device::CacheType;
+
+    #[cfg(kani)]
+    pub fn stub_cpu_count(_: u32, _: u32)->CpuidResult{
+        return CpuidResult { 
+            eax: kani::any_where(|x| *x < 10), 
+            ebx:  kani::any_where(|x| *x < 10), 
+            ecx:  kani::any_where(|x| *x < 10), 
+            edx:  kani::any_where(|x| *x < 10) 
+        }
+    }
+
+    // #[cfg(kani)]
+    // #[kani::stub(core::arch::x86_64::__cpuid_count, stub_cpu_count)]
+    // #[kani::proof]
+    // pub fn fixme_split_into_three(){ //Fixme makes Kani ignore this test for now.
+    //     let mut tree = PageRangeTree::new();
+
+    //     let addr1 = VirtAddr::new(kani::any());
+    //     let pn = PageNumber::from_address( addr1.unwrap());
+
+    //     let page = get_page();
+    //     tree.add_page(pn, page);
+
+    //     let addr2 = VirtAddr::new(kani::any());
+    //     let pn_split = PageNumber::from_address(addr2.unwrap());
+    //     let discard = kani::any();
+        
+    //     tree.split_into_three(pn_split, discard);
+
+    //     //What are our invariants
+    //     // tree.get(pn)
+        
+    // }
+
+
+    //handle page enums
+    pub fn get_page() -> Page{
+        // let pa = PhysAddr::new(kani::any()) ;
+
+        
+        let pa = PhysAddr::new(kani::any_where(|x| *x < 10));
+        //Note: kany is not generating a random
+        //value here, it will evaluate the different
+        //match states equally.
+
+        let cache_type = match kani::any(){
+            0 => CacheType::MemoryMappedIO,
+            1 => CacheType::Uncacheable,
+            2 => CacheType::WriteBack,
+            3 => CacheType::WriteCombining,
+            _ => CacheType::WriteThrough,
+        };
+
+        match kani::any(){
+            0 => Page::new(),
+            _ => Page::new_wired(pa.unwrap(), cache_type)
+        }
+    }
+
+}
+
+
```

Comment:
Is this a partial example of a kani test? Or one that's not working? If it's the former, what do we still need to do to get it to work? If it's the latter, we should not include it in the main branch, I think.